### PR TITLE
Timeouts

### LINKED ISSUE #437: Request Timeouts
If we send a message to the Avail API, but it's down or hung and we never get a response, we don't time out.  We should, perhaps with a nifty error message!

### PR DESCRIPTION
**Pull request by mboneil10**
Closes #437

### CODE REVIEW

**Comment by mboneil10** [comment]
No idea if I did this correctly.

**Comment by mboneil10** [comment]
Also, please let me know if I missed any get requests

File: src/providers/alert.service.ts [hunk 1]
```diff
@@ -12,11 +12,15 @@ export class AlertService {
...
       .then(response => response.json() as Alert[])
+      .timeout(10000)
       .catch(this.handleError);
   }
 
   private handleError(error: any): void {
-    console.error('An error occurred', error); // for demo purposes only
-    // return Promise.reject(error.message || error);
+    if (error.name === "TimeoutError") {
+      console.error('A timeout error occurred', error);
```
Comment (Anbranin) [inline src/providers/alert.service.ts]: Could you not do something like (I'm spitballing here)
``` javascript
errorMessage = 'A ' + error.name + ' error occurred'
console.error(errorMessage, error)
```
Of course, you'd have to handle whether the error.name started with a vowel or not, but personally I'd just leave out the indefinite article to get completely around that.

**Comment by mboneil10** [comment]
Don't merge yet I need to test this

**Comment by mboneil10** [comment]
GAH I messed up

**Comment by mboneil10** [comment]
Oh, the property `timeout` doesn't exist... need to figure out how to do timeouts now. It exists for`http`. Need a specific timeout for promises.

**Comment by mboneil10** [comment]
This PR doesn't look like it will get anywhere anytime soon. Closing for now.

**Pull request closed without merging**